backend/src/packages/chat-messages/chat-message.service.ts

```diff
@@ -49,6 +59,52 @@ class ChatMessageService implements Service {
     return chatMessage.toObject();
   }
 
+  public async generateReply(
+    payload: ChatMessageCreatePayload,
+  ): Promise<ChatMessageGetAllItemResponseDto> {
+    const sender = await this.chatbotService.getChatbotUser();
+    if (!sender) {
+      throw new UsersError({
+        status: HTTPCode.NOT_FOUND,
+        message: ExceptionMessage.USER_NOT_FOUND,
+      });
+    }
+
+    const chatMessages = await this.chatMessageRepository.findAllByChatId(
+      payload.chatId,
+    );
+
+    const openAiMessages: OpenAiMessageGenerateRequestDto[] = chatMessages.map(
+      (chatMessage) => {
+        const { senderId, message } = chatMessage.toObject();
+
+        return {
+          role: payload.senderId === senderId ? 'assistant' : 'user',
+          content: message,
+        };
+      },
+    );
+
+    openAiMessages.push({
+      role: 'user',
+      content: `Pretend that you a psychologist and you give mental support to the patient, don't mention that you are an AI model, be compassionate and give short answers. Answer this question: '${payload.message}'. Don't mention that you are an AI model, be compassionate and short answers.`,
```

Comment:
The last sentence doubles
```suggestion
      content: `Pretend that you a psychologist and you give mental support to the patient, don't mention that you are an AI model, be compassionate and give short answers. Answer this question: '${payload.message}'.`,
```